openai[patch] get output_type when using with_structured_output

liuhetian opened the pull request:
- This allows pydantic to correctly resolve annotations necessary when using openai new param `json_schema`

Resolves issue: #26250

liuhetian committed:
Update base.py

eyurtsev committed:
x

eyurtsev committed:
x

eyurtsev commented:
Would need to take a more careful look to see if that's the appropriate place to set the output_type... current code seems to ignore the possibility that the schema is a dict

eyurtsev commented:
@baskaryan tagging you in case this is a quick fix for you

baskaryan reviewed (approved):
awesome, thank you!

liuhetian committed:
Merge branch 'master' into master

liuhetian committed:
Update base.py

Fix type checking errors by declaring that the schema is a Pydantic BaseModel at this point.

liuhetian committed:
Update base.py

liuhetian committed:
Update base.py

liuhetian committed:
Merge branch 'master' into master

liuhetian committed:
Merge branch 'master' into master

baskaryan committed:
fix
